language/stdlib/modules/DesignatedDealer.move

```diff
@@ -5,29 +5,41 @@ module DesignatedDealer {
     use 0x1::Vector;
     use 0x1::Event;
     use 0x1::Roles;
+    use 0x1::Signer;
+    use 0x1::Coin1::Coin1;
+    use 0x1::Coin2::Coin2;
 
+    /// A `DesignatedDealer` always holds this `Dealer` resource regardless of the
+    /// currencies it can hold. All `ReceivedMintEvent` events for all
+    /// currencies will be emitted on `mint_event_handle`.
     resource struct Dealer {
+        /// Handle for mint events
+        mint_event_handle: Event::EventHandle<ReceivedMintEvent>,
+    }
+
+    /// The `TierInfo` resource holds the information needed to track which
+    /// tier a mint to a DD needs to be in.
+    resource struct TierInfo<CoinType> {
```

Comment:
Wonder if it would make sense for us to put the `Preburn` resource here in the future?